lib/netjockey/service.rb

```diff
@@ -62,16 +19,29 @@ def room(id)
       case res
       when Net::HTTPSuccess then
         data = JSON.parse(res.body)
-        room_info = RoomInfo.new(data['roomInfo']['name'],
-                                 data['roomInfo']['id'])
-        song_queue = data['queue']['playlist'].map do |song|
-          Song.new(song['duration'],
-                   song['title'], song['id'],
-                   song['thumbnailUrl'],
-                   song['uuid'])
-        end
-        Room.new(room_info, data['currentSongTime'], song_queue)
+        create_room(data)
+      when Net::HTTPNotFound then
+        raise 'room not found'
```

Comment:
What fits you best. I usually create new custom exception with name `StandardError` in namespace of my project inherited from Ruby's `StandardError`. All the other custom exceptions are inherited from this custom `StandardError` so if need be I can catch all my custom exceptions just by catching my custom `StandardError`. But that's just the way I do it, it's not a best practice or anything.